From: Artur Rona <ari-tczew@ubuntu.com>
Description: Fix parsing non-ascii filenames passed on the commandline (or through a file browser).
Bug: http://code.google.com/p/clementine-player/issues/detail?id=752
Bug-Ubuntu: https://launchpad.net/bugs/694423
Origin: upstream, http://code.google.com/p/clementine-player/source/detail?r=2498
Forwarded: not-needed
Author: David Sansome <me@davidsansome.com>

diff -pruN -x '*~' clementine-0.6.orig/src/main.cpp clementine-0.6/src/main.cpp
--- clementine-0.6.orig/src/main.cpp	2010-12-26 01:11:40.000000000 +0100
+++ clementine-0.6/src/main.cpp	2010-12-26 21:37:39.000000000 +0100
@@ -154,16 +154,20 @@ int main(int argc, char *argv[]) {
   lastfm::ws::ApiKey = LastFMService::kApiKey;
   lastfm::ws::SharedSecret = LastFMService::kSecret;
 
-  // Parse commandline options - need to do this before starting the
-  // QApplication so it works without an X server
   CommandlineOptions options(argc, argv);
-  if (!options.Parse())
-    return 1;
 
   {
     // Only start a core application now so we can check if there's another
     // Clementine running without needing an X server.
+    // This MUST be done before parsing the commandline options so QTextCodec
+    // gets the right system locale for filenames.
     QtSingleCoreApplication a(argc, argv);
+
+    // Parse commandline options - need to do this before starting the
+    // full QApplication so it works without an X server
+    if (!options.Parse())
+      return 1;
+
     if (a.isRunning()) {
       if (options.is_empty()) {
         qDebug() << "Clementine is already running - activating existing window";
